server/src/test/java/org/elasticsearch/health/HealthPeriodicLoggerTests.java

```diff
@@ -203,60 +223,119 @@ public void testEnabled() {
+                Settings.builder()
+                    .put(HealthPeriodicLogger.POLL_INTERVAL_SETTING.getKey(), pollInterval)
+                    // Since the default value of enabled is false, if we do not set it here it disable it
+                    .put(HealthPeriodicLogger.ENABLED_SETTING.getKey(), true)
+                    .build()
+            );
+            assertEquals(pollInterval, testHealthPeriodicLogger.getPollInterval());
+            verifyLoggerIsReadyToRun(testHealthPeriodicLogger);
+            assertNotNull(testHealthPeriodicLogger.getScheduler());
+            assertTrue(
+                testHealthPeriodicLogger.getScheduler().scheduledJobIds().contains(HealthPeriodicLogger.HEALTH_PERIODIC_LOGGER_JOB_NAME)
+            );
+        }
 
-            @Override
-            public void onFailure(Exception e) {
-                failureCalled.set(true);
-            }
-        };
-        HealthService testHealthService = this.getMockedHealthService();
+        // Poll interval doesn't schedule a job when disabled
+        {
+            TimeValue pollInterval = TimeValue.timeValueSeconds(randomIntBetween(15, 59));
+            this.clusterSettings.applySettings(
+                Settings.builder()
+                    .put(HealthPeriodicLogger.POLL_INTERVAL_SETTING.getKey(), pollInterval)
+                    .put(HealthPeriodicLogger.ENABLED_SETTING.getKey(), false)
+                    .build()
+            );
+            assertFalse(testHealthPeriodicLogger.enabled());
+            assertEquals(pollInterval, testHealthPeriodicLogger.getPollInterval());
+            assertFalse(
+                testHealthPeriodicLogger.getScheduler().scheduledJobIds().contains(HealthPeriodicLogger.HEALTH_PERIODIC_LOGGER_JOB_NAME)
+            );
+            // Re-enable
+            this.clusterSettings.applySettings(
+                Settings.builder().put(HealthPeriodicLogger.POLL_INTERVAL_SETTING.getKey(), pollInterval).build()
+            );
+        }
 
-        testHealthPeriodicLogger = createAndInitHealthPeriodicLogger(this.clusterService, testHealthService, true);
+        testHealthPeriodicLogger.stop();
+        // verify that updating the polling interval doesn't schedule the job if it's stopped
+        {
+            this.clusterSettings.applySettings(
+                Settings.builder()
+                    .put(HealthPeriodicLogger.POLL_INTERVAL_SETTING.getKey(), TimeValue.timeValueSeconds(30))
+                    .put(HealthPeriodicLogger.ENABLED_SETTING.getKey(), true)
+                    .build()
+            );
+            assertTrue("health logger should be enabled", testHealthPeriodicLogger.enabled());
+            assertFalse(
+                testHealthPeriodicLogger.getScheduler().scheduledJobIds().contains(HealthPeriodicLogger.HEALTH_PERIODIC_LOGGER_JOB_NAME)
+            );
+        }
+    }
 
-        HealthPeriodicLogger spyHealthPeriodicLogger = spy(testHealthPeriodicLogger);
```

Comment:
If this is increasing the difficulty of reviewing the PR, I can extract it to a different PR. I totally missed when I reviewd this but I strongly believe we should not use `spy(...)` because according to the docs:

> Creates a spy of the real object. The spy calls real methods unless they are stubbed.

This means we are not testing the `tryToLogHealth` at all. This felt pretty important to me, that's why I added it here.